### PR TITLE
Add ;magic (untested)

### PR DESCRIPTION
**Pull request by dayvonjersen**
literal copy/paste job from cute/cute.c 
`sed -i 's/cute/magic/g'` :shipit:

### CODE REVIEW

File: mods/magic/magic.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#include "../../core/irc.h"
+
+/* yes, these are stolen from glitz,
+ * they also use the same format to steal more in the future */
+static char *target_magics[] = {
+    "(つ˵•́ω•̀˵)つ━☆ﾟ.*･｡ﾟ҉̛ {target}",
+    "(つ˵•́ω•̀˵)つ━☆✿✿✿✿✿✿ {target}",
+    "╰( ´・ω・)つ──☆ﾟ.*･｡ {target}",
+    "╰( ´・ω・)つ──☆✿✿✿✿✿✿ {target}",
+    "(○´･∀･)o<･。:*ﾟ;+．{target}",
```
Comment (dayvonjersen) [inline mods/magic/magic.c:23]: trailing comma in initializer list probably isn't a thing in C

File: mods/magic/magic.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+static char *magics[] = {
+    "pew pew mf :DDD"
```
Comment (dayvonjersen) [inline mods/magic/magic.c:27]: but if it is a thing then a comma is needed here

File: mods/magic/magic.c [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	if (strncmp(msg, "magic", 4)) {
+		return;
+	}
+
+	srand(time(NULL));
+	char magic_pick[BUFSIZE] = {'\0'};
+	char *target = strchr(msg, ' ');
+
+	/* prepare the magic, targeted or not */
+	strcpy(magic_pick, target ? target_magics[rand()%13] : magics[rand()%35]);
```
Comment (dayvonjersen) [inline mods/magic/magic.c:42]: whoops didn't change these numbers 🙃👍

**Comment by Ckath** [comment]
lets see

File: mods/magic/magic.c (hunk 3 above)
Comment (Ckath) [reply mods/magic/magic.c:42]: better fix the strncmp and the rand range then its probably non segfaulting

**Review by Ckath — CHANGES_REQUESTED** [review]
just that one thing really

**Review by Ckath — APPROVED** [review]

**Pull request merged**